Litellm gemini batch

FelipeRodriguesGare opened the pull request:
## Title

Implement file_content handler for vertex_ai provider and fix create_file.

## Relevant issues

Vertex_ai batch endpoint was not working properly. The create_file was uploading an array instead of an .jsonl file. The file_content handler did not exist making it not possible to retrieve batch file response.

## Pre-Submission checklist
<img width="1283" height="322" alt="image" src="https://github.com/user-attachments/assets/2a0df332-76a3-4a56-b20c-00e4c8baf593" />
**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem



## Type

🆕 New Feature
🐛 Bug Fix
✅ Test

## Changes
- Fixed create_file for vertex_ai
- Added handler for file_content for vertex_ai

FelipeRodriguesGare committed:
feat: add Vertex AI support for file content retrieval

- Extended `custom_llm_provider` to include "vertex_ai" in `afile_content` function.
- Implemented file content retrieval logic for Vertex AI in `VertexAIFilesHandler`.
- Added helper method to extract bucket and object from URL-encoded file_id.
- Created comprehensive unit and integration tests for Vertex AI file handling.
- Updated transformation logic to ensure compatibility with Vertex AI file responses.

FelipeRodriguesGare committed:
fix: update Vertex AI file transformation logic

- Modified the transformation logic in `VertexAIFilesConfig` to return a newline-separated JSON string for batch JSONL files instead of a array if JSON strings.

ishaan-jaff reviewed (changes requested):
@FelipeRodriguesGare can you add screenshots to show this worked after manual testing ?

FelipeRodriguesGare commented:
Sure, here it is:

The create file (POST - /files)
<img width="2340" height="642" alt="image" src="https://github.com/user-attachments/assets/d8028074-feb1-4a26-9cc4-2b2960275e47" />

The batch creation (POST - /batches)
<img width="2342" height="1016" alt="image" src="https://github.com/user-attachments/assets/d94664e8-6a22-4d41-8e01-c93587625d36" />

The retrieve of the batch (GET - /batch/{id})
<img width="2338" height="1042" alt="image" src="https://github.com/user-attachments/assets/b29f5917-ddcd-4646-a603-3c9d8042ded8" />

The retrieve of the file (GET - /file/{id}/content)
<img width="2326" height="1078" alt="image" src="https://github.com/user-attachments/assets/3fb2292a-865b-4d3a-bb1d-814292f0e1fb" />

FelipeRodriguesGare committed:
fix: enhance Vertex AI output handling in transformation logic

- Updated the transformation logic in `VertexAIBatchTransformation` to utilize the new `OutputInfo` TypedDict for retrieving the GCS output directory.
- Added `OutputInfo` class to type definitions for better structure and clarity in Vertex AI responses.

ishaan-jaff reviewed (approved):
LGTM @FelipeRodriguesGare would you be interested in contributing a guide on how to use VertexAI Batch APIs with LiteLLM ? 

e.g as this one here https://docs.litellm.ai/docs/providers/bedrock_batches
